modules/schedule_module/php/modulerow.class.inc

```diff
@@ -0,0 +1,117 @@
+ *
+ * @category   Core
+ * @package    Main
+ * @subpackage Core
+ * @author     Shen Wang <wangshen.mcin@gmail.com>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class ModuleRow implements \LORIS\Data\DataInstance
+{
+    protected $AppointmentID;
+    protected $SessionID;
+    protected $AppointmentTypeID;
+    protected $StartsAt;
+    protected $Day;
+    protected $Time;
+    protected $PSCID;
+    protected $DCCID;
+    protected $Visit_label;
+    protected $CenterID;
+    protected $SubprojectID;
+    protected $Project;
+    protected $Subproject;
+    protected $started;
+    protected $AppointmentTypeName;
+    protected $site;
+    protected $ProjectID;
+    /**
+     * Create a new ModuleRow Instance.
+     *
+     * @param array $row The ModuleRow Instance
+     */
+    public function __construct(array $row)
+    {
+        $this->AppointmentID     = $row['AppointmentID'];
+        $this->SessionID         = $row['SessionID'];
+        $this->AppointmentTypeID = $row['AppointmentTypeID'];
+        $this->Day          = $row['day'];
+        $this->Time         = substr($row['time'], 0, -3);
+        $this->PSCID        = $row["PSCID"];
+        $this->DCCID        = $row['DCCID'];
+        $this->Visit_label  = $row['Visit_label'];
+        $this->CenterID     = $row['CenterID'];
+        $this->SubprojectID = $row['CohortID'];
+        $this->Project      = $row['Project'];
+        $this->Subproject   = $row['title'];
+        $this->StartsAt     = $row['StartsAt'];
+        $this->AppointmentTypeName = $row['AppointmentTypeName'];
+        $this->site = $row['Name'];
+    }
+
+    /**
+     * Implements \LORIS\Data\DataInstance interface for this row.
+     *
+     * @return array the instance data.
+     */
+    public function jsonSerialize() : array
+    {
+        return
+            [
+                'DCCID'               => $this->DCCID,
+                'PSCID'               => $this->PSCID,
+                'site'                => $this->site,
+                'VisitLabel'          => $this->Visit_label,
+                'Project'             => $this->Project,
+                'Subproject'          => $this->Subproject,
+                'AppointmentTypeID'   =>$this->AppointmentTypeID,
+                'Day'                 => $this->Day,
+                'Time'                => $this->Time,
+                'StartsAt'            =>$this->StartsAt,
+                'SessionID'           =>$this->SessionID,
+                'AppointmentID'       =>$this->AppointmentID,
+                'AppointmentTypeName' => $this->AppointmentTypeName,
+                'CenterID'            => $this->CenterID,
+                'SubprojectID'        => $this->SubprojectID,
+            ];
+    }
+    /**
+     * Returns the CenterID for this row, for filters such as
+     * \LORIS\Data\Filters\UserSiteMatch to match again.
```

Comment:
The UserSiteMatch filter checks that the SiteHaver interface is implemented, which doesn't seem to be the case for this class (it's not in the list of the implements at the start.)

The return type for this function is also wrong for the interface (it needs to return a `\CenterID`, not an int.)